src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/LongPollingTransport.java

```diff
@@ -0,0 +1,123 @@
+    private OnReceiveCallBack onReceiveCallBack;
+    private TransportOnClosedCallback onClose;
+    private String url;
+    private final HttpClient client;
+    private final HttpClient pollingClient;
+    private final Map<String, String> headers;
+    private static final int POLL_TIMEOUT = 100*1000;
+    private volatile Boolean active;
+    private String pollUrl;
+    private String closeError;
+    private CompletableSubject pollCompletableSubject = CompletableSubject.create();
+    private final Logger logger = LoggerFactory.getLogger(LongPollingTransport.class);
+
+    public LongPollingTransport(Map<String, String> headers, HttpClient client) {
+        this.headers = headers;
+        this.client = client;
+        this.pollingClient = client.cloneWithTimeOut(POLL_TIMEOUT);
+    }
+
+    //Package private active accessor for testing.
+    boolean isActive(){
+        return this.active;
+    }
+
+    @Override
+    public Completable start(String url) {
+        this.active = true;
+        logger.info("Starting LongPolling transport");
+        this.url = url;
+        pollUrl = url + "&_=" + System.currentTimeMillis();
+        logger.info("Polling {}", pollUrl);
+        HttpRequest request = new HttpRequest();
+        request.addHeaders(headers);
+        return this.pollingClient.get(pollUrl, request).flatMapCompletable(response -> {
+            if (response.getStatusCode() != 200){
+                logger.error("Unexpected response code {}", response.getStatusCode());
+                this.active = false;
+                return Completable.error(new Exception("Failed to connect."));
+            } else {
+                logger.info("Activating poll loop", response.getStatusCode());
+                this.active = true;
+            }
+            poll(url).subscribeWith(pollCompletableSubject);
+
+            return Completable.complete();
+        });
+    }
+
+    private Completable poll(String url){
+        if (this.active) {
+            pollUrl = url + "&_=" + System.currentTimeMillis();
+            logger.info("Polling {}", pollUrl);
+            HttpRequest request = new HttpRequest();
+            request.addHeaders(headers);
+            Completable pollingCompletable = this.pollingClient.get(pollUrl, request).flatMapCompletable(response -> {
+                if (response.getStatusCode() == 204) {
+                    logger.info("LongPolling transport terminated by server.");
+                    this.active = false;
+                } else if (response.getStatusCode() != 200) {
+                    logger.error("Unexpected response code {}", response.getStatusCode());
+                    this.active = false;
+                    this.closeError = "Unexpected response code " + response.getStatusCode();
+                } else {
+                    logger.info("Message received");
+                    new Thread(() -> this.onReceive(response.getContent())).start();
```

Comment:
Hmm do you have more insight on this @JonathanGiles ?

Reply:
You probably don't want to do this. You probably want to introduce some thread pooling so that you don't overwhelm the app / system with potentially thousands of threads. Look into Executors / ExecutorService.

Reply:
Thank you! We always appreciate your insight!